ims/database/image.py

```diff
@@ -137,12 +137,21 @@ def move_image(self, src_project_name, name, dest_project_id,
     # returns a array of image ids of the images which have the given name
     @log
     def fetch_id_with_name_from_project(self, name, project_name):
+        """
+        searches for image by name and returns image id
+
+        :param  name: name of the image
+        :poaram project_name: name of the project
+        :return: image.id: the image id of the image
+        """
+
         try:
             image = self.connection.session.query(Image). \
                 filter(Image.project.has(name=project_name)).filter_by(
                 name=name).one_or_none()
-            if image is not None:
-                return image.id
+            if image is None:
+                raise db_exceptions.ImageNotFoundException(name)
+            return image.id
```

Comment:
I mean it could only potentially be a bug, it's actually fine now. It's just good practice is all for the bug I mentioned above. It was just a suggestion though, so feel free to ignore it.